In [6]:
report_card_dictionary = {"student_names": ["Nishika","Aakriti","Aayushi","Peony","Riya"],
                        "physics_grades": [34, 45, 56, 78, 100],
                        "chemistry_grades": [100, 78, 0, 45, 98],
                        "math_grades": [56, 76, 34, 23, 97],
                        "biology_grades": [45, 56, 34, 67, 78]
                        }
report_card_dictionary

{'student_names': ['Nishika', 'Aakriti', 'Aayushi', 'Peony', 'Riya'],
 'physics_grades': [34, 45, 56, 78, 100],
 'chemistry_grades': [100, 78, 0, 45, 98],
 'math_grades': [56, 76, 34, 23, 97],
 'biology_grades': [45, 56, 34, 67, 78]}

In [7]:
#setup
import pandas as pd

In [8]:
#making the report card dictionary into a dataframe
report_card_df = pd.DataFrame(report_card_dictionary)
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0       Nishika              34               100           56              45
1       Aakriti              45                78           76              56
2       Aayushi              56                 0           34              34
3         Peony              78                45           23              67
4          Riya             100                98           97              78

In [9]:
#first 3 rows in the report card dataframe 
report_card_df.head(3)

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0       Nishika              34               100           56              45
1       Aakriti              45                78           76              56
2       Aayushi              56                 0           34              34

In [10]:
# 3 random rows in the report card dataframe
report_card_df.sample(3)

student_names  physics_grades  chemistry_grades  math_grades  biology_grades
0       Nishika              34               100           56              45
2       Aayushi              56                 0           34              34
1       Aakriti              45                78           76              56

In [11]:
#describing all the numeric column in the report card dataframe 
report_card_df.describe()

physics_grades  chemistry_grades  math_grades  biology_grades
count        5.000000          5.000000      5.00000        5.000000
mean        62.600000         64.200000     57.20000       56.000000
std         26.491508         42.144988     30.19437       17.392527
min         34.000000          0.000000     23.00000       34.000000
25%         45.000000         45.000000     34.00000       45.000000
50%         56.000000         78.000000     56.00000       56.000000
75%         78.000000         98.000000     76.00000       67.000000
max        100.000000        100.000000     97.00000       78.000000

#insights
- average chemistry grade? 64.2
- average biology grade? 56.0
- lowest math grade? 23.0
- highest physics grade? 100
- scores preesent in the math column? 5

In [17]:
report_card_df["total_grades"] = report_card_df["physics_grades"]+ report_card_df["math_grades"]+ report_card_df["chemistry_grades"]+ report_card_df["biology_grades"]
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0       Nishika              34               100           56   
1       Aakriti              45                78           76   
2       Aayushi              56                 0           34   
3         Peony              78                45           23   
4          Riya             100                98           97   

   biology_grades  total_grades  
0              45           235  
1              56           255  
2              34           124  
3              67           213  
4              78           373

In [18]:
report_card_df["avg_grades"] = report_card_df["total_grades"]/4
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0       Nishika              34               100           56   
1       Aakriti              45                78           76   
2       Aayushi              56                 0           34   
3         Peony              78                45           23   
4          Riya             100                98           97   

   biology_grades  total_grades  avg_grades  
0              45           235       58.75  
1              56           255       63.75  
2              34           124       31.00  
3              67           213       53.25  
4              78           373       93.25

In [19]:
report_card_insights_df = report_card_df.describe().transpose()
report_card_insights_df

count   mean        std    min     25%     50%     75%  \
physics_grades      5.0   62.6  26.491508   34.0   45.00   56.00   78.00   
chemistry_grades    5.0   64.2  42.144988    0.0   45.00   78.00   98.00   
math_grades         5.0   57.2  30.194370   23.0   34.00   56.00   76.00   
biology_grades      5.0   56.0  17.392527   34.0   45.00   56.00   67.00   
total_grades        5.0  240.0  89.615847  124.0  213.00  235.00  255.00   
avg_grades          5.0   60.0  22.403962   31.0   53.25   58.75   63.75   

                     max  
physics_grades    100.00  
chemistry_grades  100.00  
math_grades        97.00  
biology_grades     78.00  
total_grades      373.00  
avg_grades         93.25

In [20]:
#saving a csv file 
report_card_insights_df.to_csv("report card insights.csv")

REPORT CARD ALPHABET GRADES

In [21]:
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0       Nishika              34               100           56   
1       Aakriti              45                78           76   
2       Aayushi              56                 0           34   
3         Peony              78                45           23   
4          Riya             100                98           97   

   biology_grades  total_grades  avg_grades  
0              45           235       58.75  
1              56           255       63.75  
2              34           124       31.00  
3              67           213       53.25  
4              78           373       93.25

In [24]:
report_card_df.loc[report_card_df["total_grades"]>=140, "Status"]= "Pass"
report_card_df.loc[report_card_df["total_grades"]<140, "Status"]= "Fail"
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0       Nishika              34               100           56   
1       Aakriti              45                78           76   
2       Aayushi              56                 0           34   
3         Peony              78                45           23   
4          Riya             100                98           97   

   biology_grades  total_grades  avg_grades Status  
0              45           235       58.75   Pass  
1              56           255       63.75   Pass  
2              34           124       31.00   Fail  
3              67           213       53.25   Pass  
4              78           373       93.25   Pass

In [26]:
#making an alphabet grade column 
report_card_df.loc[(report_card_df["avg_grades"] >=90), "Character_Grade"]= "A"
report_card_df.loc[(report_card_df["avg_grades"] >=70) & (report_card_df["avg_grades"]<90), "Character_grade"]= "B"
report_card_df.loc[(report_card_df["avg_grades"] >=50) & (report_card_df["avg_grades"]<70), "Character_grade"]= "C"
report_card_df.loc[(report_card_df["avg_grades"] >=35) & (report_card_df["avg_grades"]<50), "Character_grade"]= "D"
report_card_df.loc[(report_card_df["avg_grades"] <35), "Character_grade"]= "F"

In [27]:
report_card_df

student_names  physics_grades  chemistry_grades  math_grades  \
0       Nishika              34               100           56   
1       Aakriti              45                78           76   
2       Aayushi              56                 0           34   
3         Peony              78                45           23   
4          Riya             100                98           97   

   biology_grades  total_grades  avg_grades Status Character_Grade  \
0              45           235       58.75   Pass             NaN   
1              56           255       63.75   Pass             NaN   
2              34           124       31.00   Fail             NaN   
3              67           213       53.25   Pass             NaN   
4              78           373       93.25   Pass               A   

  Character_grade  
0               C  
1               C  
2               F  
3               C  
4             NaN